include/oxen/quic/ip.hpp

```diff
@@ -7,36 +7,45 @@
 
 namespace oxen::quic
 {
+    inline constexpr size_t IPV4_ARRLEN{4};
+    inline constexpr size_t IPV6_ARRLEN{16};
+
     struct ipv4
     {
-        // Stored in network order
+        // host order
         uint32_t addr;
 
-        ipv4() = default;
-        // Constructor using host order IP address string
-        ipv4(const std::string& ip);
+        constexpr ipv4() = default;
 
         constexpr ipv4(uint32_t a) : addr{a} {}
         constexpr ipv4(uint8_t a, uint8_t b, uint8_t c, uint8_t d) :
                 ipv4{uint32_t{a} << 24 | uint32_t{b} << 16 | uint32_t{c} << 8 | uint32_t{d}}
         {}
 
+        constexpr std::optional<ipv4> next_ip() const
+        {
+            std::optional<ipv4> ret = std::nullopt;
+
+            if (not will_overflow(addr))
+                ret = ipv4{addr + 1};
+
+            return ret;
```

Comment:
This could be a bit simpler if you just ditched the `will_overflow` and just detect overflow:
```suggestion
            auto next_addr = addr + 1;
            if (next_addr == 0)
                return std::nullopt;
            return ipv4{next_addr};
```